### PR TITLE
ADEN-4281 | call Rubicon on each PV on Mercury

### PR DESCRIPTION
**Pull request by bognix**

### CODE REVIEW

**Comment by WikiaTech** [comment]
SonarQube analysis reported 9 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [AdEngine2Hooks.class.php#L23](https://github.com/Wikia/app/blob/47038efd0f069333177899f439d6714f71662398/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L23): Remove the unused function parameters "$title, $article, $output, $user, $wiki". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [AdEngine2Hooks.class.php#L94](https://github.com/Wikia/app/blob/47038efd0f069333177899f439d6714f71662398/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L94): Remove the unused function parameter "$scripts". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [AdEngine2Hooks.class.php#L195](https://github.com/Wikia/app/blob/47038efd0f069333177899f439d6714f71662398/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L195): Remove the unused function parameter "$skin". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [AdEngine2Hooks.class.php#L220](https://github.com/Wikia/app/blob/47038efd0f069333177899f439d6714f71662398/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L220): Remove the unused function parameters "$jsExtensionPackages, $scssPackages". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [mercury.run.js#L17](https://github.com/Wikia/app/blob/47038efd0f069333177899f439d6714f71662398/extensions/wikia/AdEngine/js/run/mercury.run.js#L17): Function has 14 parameters which is greater than 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. INFO [AdEngine2Hooks.class.php#L26](https://github.com/Wikia/app/blob/47038efd0f069333177899f439d6714f71662398/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L26): Complete the task associated to this "TODO" comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1135)
1. INFO [AdEngine2Hooks.class.php#L109](https://github.com/Wikia/app/blob/47038efd0f069333177899f439d6714f71662398/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L109): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)
1. INFO [AdEngine2Hooks.class.php#L110](https://github.com/Wikia/app/blob/47038efd0f069333177899f439d6714f71662398/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L110): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)
1. INFO [AdEngine2Hooks.class.php#L111](https://github.com/Wikia/app/blob/47038efd0f069333177899f439d6714f71662398/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L111): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 6 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [AdEngine2Hooks.class.php#L23](https://github.com/Wikia/app/blob/52007366023898f367bc05236fed6ea61bf1cca7/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L23): Remove the unused function parameters "$title, $article, $output, $user, $wiki". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [AdEngine2Hooks.class.php#L94](https://github.com/Wikia/app/blob/52007366023898f367bc05236fed6ea61bf1cca7/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L94): Remove the unused function parameter "$scripts". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [AdEngine2Hooks.class.php#L198](https://github.com/Wikia/app/blob/52007366023898f367bc05236fed6ea61bf1cca7/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L198): Remove the unused function parameter "$skin". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [AdEngine2Hooks.class.php#L223](https://github.com/Wikia/app/blob/52007366023898f367bc05236fed6ea61bf1cca7/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L223): Remove the unused function parameters "$jsExtensionPackages, $scssPackages". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [mercury.run.js#L17](https://github.com/Wikia/app/blob/52007366023898f367bc05236fed6ea61bf1cca7/extensions/wikia/AdEngine/js/run/mercury.run.js#L17): Function has 14 parameters which is greater than 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. INFO [AdEngine2Hooks.class.php#L26](https://github.com/Wikia/app/blob/52007366023898f367bc05236fed6ea61bf1cca7/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L26): Complete the task associated to this "TODO" comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1135)

File: extensions/wikia/AdEngine/js/lookup/rubicon/rubiconFastlane.js [hunk 1]
```diff
@@ -222,20 +224,25 @@ define('ext.wikia.adEngine.lookup.rubicon.rubiconFastlane', [
...
-		rubicon.type = 'text/javascript';
-		rubicon.src = rubiconLibraryUrl;
+			rubicon.async = true;
+			rubicon.type = 'text/javascript';
+			rubicon.src = rubiconLibraryUrl;
+
+			node.parentNode.insertBefore(rubicon, node);
+			context = adContext.getContext();
+			configureSlots(skin);
+			response = false;
```
Comment (fraszczakszymon) [inline extensions/wikia/AdEngine/js/lookup/rubicon/rubiconFastlane.js]: Shouldn't we reset this value on each pv?

File: extensions/wikia/AdEngine/js/run/mercury.run.js [hunk 2]
```diff
@@ -67,4 +67,15 @@ require([
...
+	if (
+		geo.isProperGeo(instantGlobals.wgAdDriverRubiconFastlaneCountries) &&
+		geo.isProperGeo(instantGlobals.wgAdDriverRubiconFastlaneMercuryFixCountries)
+	) {
+		mercuryListener.onEveryPageChange(function () {
+			rubiconFastlane.call();
+		});
+	}
+
+
```
Comment (fraszczakszymon) [inline extensions/wikia/AdEngine/js/run/mercury.run.js]: 🙈 (lines)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 6 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [AdEngine2Hooks.class.php#L23](https://github.com/Wikia/app/blob/9a20163fecaee1c79a6e328a3624f627711c8914/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L23): Remove the unused function parameters "$title, $article, $output, $user, $wiki". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [AdEngine2Hooks.class.php#L94](https://github.com/Wikia/app/blob/9a20163fecaee1c79a6e328a3624f627711c8914/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L94): Remove the unused function parameter "$scripts". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [AdEngine2Hooks.class.php#L198](https://github.com/Wikia/app/blob/9a20163fecaee1c79a6e328a3624f627711c8914/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L198): Remove the unused function parameter "$skin". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [AdEngine2Hooks.class.php#L223](https://github.com/Wikia/app/blob/9a20163fecaee1c79a6e328a3624f627711c8914/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L223): Remove the unused function parameters "$jsExtensionPackages, $scssPackages". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [mercury.run.js#L17](https://github.com/Wikia/app/blob/9a20163fecaee1c79a6e328a3624f627711c8914/extensions/wikia/AdEngine/js/run/mercury.run.js#L17): Function has 14 parameters which is greater than 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. INFO [AdEngine2Hooks.class.php#L26](https://github.com/Wikia/app/blob/9a20163fecaee1c79a6e328a3624f627711c8914/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L26): Complete the task associated to this "TODO" comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1135)

**Review by fraszczakszymon — APPROVED** [review]
LGTM 🚀

**Comment by WikiaTech** [comment]
SonarQube analysis reported 6 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [AdEngine2Hooks.class.php#L23](https://github.com/Wikia/app/blob/86bded52aa8b0723ff52e1a00202d2b808bbd4b9/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L23): Remove the unused function parameters "$title, $article, $output, $user, $wiki". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [AdEngine2Hooks.class.php#L94](https://github.com/Wikia/app/blob/86bded52aa8b0723ff52e1a00202d2b808bbd4b9/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L94): Remove the unused function parameter "$scripts". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [AdEngine2Hooks.class.php#L198](https://github.com/Wikia/app/blob/86bded52aa8b0723ff52e1a00202d2b808bbd4b9/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L198): Remove the unused function parameter "$skin". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [AdEngine2Hooks.class.php#L223](https://github.com/Wikia/app/blob/86bded52aa8b0723ff52e1a00202d2b808bbd4b9/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L223): Remove the unused function parameters "$jsExtensionPackages, $scssPackages". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [mercury.run.js#L17](https://github.com/Wikia/app/blob/86bded52aa8b0723ff52e1a00202d2b808bbd4b9/extensions/wikia/AdEngine/js/run/mercury.run.js#L17): Function has 14 parameters which is greater than 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. INFO [AdEngine2Hooks.class.php#L26](https://github.com/Wikia/app/blob/86bded52aa8b0723ff52e1a00202d2b808bbd4b9/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L26): Complete the task associated to this "TODO" comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1135)

**Pull request merged**